llms/anthropic/internal/anthropicclient/messages.go

```diff
@@ -0,0 +1,297 @@
+	}
+	return response, nil
+}
+
+func handleMessageStartEvent(event map[string]interface{}, response MessageResponsePayload) (MessageResponsePayload, error) {
+	message, ok := event["message"].(map[string]interface{})
+	if !ok {
+		return response, errors.New("invalid message field type")
+	}
+
+	usage, ok := message["usage"].(map[string]interface{})
+	if !ok {
+		return response, errors.New("invalid usage field type")
+	}
+
+	inputTokens, err := getFloat64(usage, "input_tokens")
+	if err != nil {
+		return response, err
+	}
+
+	response.ID = getString(message, "id")
+	response.Model = getString(message, "model")
+	response.Role = getString(message, "role")
+	response.Type = getString(message, "type")
+	response.Usage.InputTokens = int(inputTokens)
+
+	return response, nil
+}
+
+func handleContentBlockStartEvent(event map[string]interface{}, response MessageResponsePayload) (MessageResponsePayload, error) {
+	indexValue, ok := event["index"].(float64)
+	if !ok {
+		return response, errors.New("invalid index field type")
+	}
+	index := int(indexValue)
+
+	if len(response.Content) <= index {
+		response.Content = append(response.Content, struct {
+			Text string `json:"text"`
+			Type string `json:"type"`
+		}{})
+	}
+	return response, nil
+}
+
+func handleContentBlockDeltaEvent(ctx context.Context, event map[string]interface{}, response MessageResponsePayload, payload *messagePayload) (MessageResponsePayload, error) {
+	indexValue, ok := event["index"].(float64)
+	if !ok {
+		return response, errors.New("invalid index field type")
+	}
+	index := int(indexValue)
+
+	delta, ok := event["delta"].(map[string]interface{})
+	if !ok {
+		return response, errors.New("invalid delta field type")
+	}
+	deltaType, ok := delta["type"].(string)
+	if !ok {
+		return response, errors.New("invalid delta type field type")
+	}
+
+	if deltaType == "text_delta" {
+		text, ok := delta["text"].(string)
+		if !ok {
+			return response, errors.New("invalid delta text field type")
+		}
+		if len(response.Content) > index {
+			response.Content[index].Text += text
+		} else {
+			return response, errors.New("content index out of range")
+		}
+	}
+
+	if payload.StreamingFunc != nil {
+		text, ok := delta["text"].(string)
+		if !ok {
+			return response, errors.New("invalid delta text field type")
+		}
+		err := payload.StreamingFunc(ctx, []byte(text))
+		if err != nil {
+			return response, fmt.Errorf("streaming func returned an error: %w", err)
+		}
+	}
+	return response, nil
+}
+
+func handleMessageDeltaEvent(event map[string]interface{}, response MessageResponsePayload) (MessageResponsePayload, error) {
+	delta, ok := event["delta"].(map[string]interface{})
+	if !ok {
+		return response, errors.New("invalid delta field type")
+	}
+	if stopReason, ok := delta["stop_reason"].(string); ok {
+		response.StopReason = stopReason
+	}
+
+	usage, ok := event["usage"].(map[string]interface{})
+	if !ok {
+		return response, errors.New("invalid usage field type")
```

Comment:
while an internal package, generally it's better to define these errors up top so calling code can use errors.Is checks.